### PR TITLE
mkcloud: refactor setuphost

### PR DESCRIPTION
**Pull request by bmwiedemann**
add tests and several fixes

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -4130,6 +4130,27 @@ function oncontroller_prepare_functional_tests
...
+    export SHAREDVG=1
+    export debug_mkcloud=1
+    # must be different from our admin net
+    # and in a private range to make libvirt happy
+    export net_admin=172.27.176
+    export admin_node_memory=1048576
+    unset adminip
+    rm -f $mkcconf
+    onadmin_setup_local_zypper_repositories
+    ~/scripts/mkcloud setuphost cleanup prepare setupadmin
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: safely

Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: $SCRIPTS_DIR

File: scripts/lib/mkcloud-common.sh [hunk 2]
```diff
@@ -346,11 +346,13 @@ function jsonice
 
 function ensure_packages_installed
 {
-    local zypper_params="--non-interactive --gpg-auto-import-keys --no-gpg-checks"
-    local pack
-    for pack in "$@" ; do
-        rpm -q $pack &> /dev/null || safely zypper $zypper_params install "$pack"
-    done
+    if is_suse ; then
+        export ZYPP_LOCK_TIMEOUT=60
```
Comment (aplanas) [inline scripts/lib/mkcloud-common.sh]: +1

File: scripts/mkcloud [hunk 3]
```diff
@@ -1308,6 +1309,9 @@ function expand_steps
                     _testupdate|testupdate)
                         runcmds="$runcmds addupdaterepo runupdate testsetup cct"
                     ;;
+                    _test_setuphost)
+                        runcmds="$runcmds cleanup prepare setupadmin onadmin+test_setuphost"
```
Comment (aplanas) [inline scripts/mkcloud]: onadmin_test_setuphost?

Comment (bmwiedemann) [inline scripts/mkcloud]: nay, this is calling the "onadmin" function with parameter "test_setuphost" which in the end will run the onadmin_test_setuphost function in the admin VM

**Pull request merged**